pkg/io/binaryReader.go

```diff
@@ -82,25 +77,16 @@ func (r *BinReader) ReadArray(t interface{}, maxSize ...int) {
 		} else {
 			elem = arr.Index(i).Addr()
 		}
-		method := elem.MethodByName("DecodeBinary")
-		method.Call([]reflect.Value{reflect.ValueOf(r)})
-	}
 
-	value.Elem().Set(arr)
-}
+		el, ok := elem.Interface().(decodable)
```

Comment:
I'm still not sure it's worth making these decodable/encodable interfaces. I understand that it resembles `io.Reader`/`io.Writer` and seems nice wrt to `io` symmetry, but if we're making them internal to the package they become a copy-paste of `Serializable`. If you're to make them exported then they're almost useless to the outside user, because we don't have any type that wants to implement decoder without encoder or the other way around. So maybe plain and simple `.(Serializable)` will do? This package is built around this type, after all.

Reply:
There are a certain contexts where an item can be encoded, but cant be decoded.
E.g. serialization of `[]util.Uint256` via `WriteArray` will panic if we will cast to `Serializable`.